### PR TITLE
Get GPS location of agent if change in distance is 1 m.

### PR DESCRIPTION
**Pull request by Pragya1805**
Added functionality of updating GPS location of patient in agent application and printing it on terminal if location changes by 1 m.  Space between /// and start of comment taken care of, typos as pointed out in PR for frontend_agent_verification are corrected in this branch as well.

### CODE REVIEW

File: frontend/gpay_agent/lib/views/welcome_agent.dart [hunk 1]
```diff
@@ -78,8 +78,8 @@ class WelcomeAgent extends StatelessWidget{
     );
   }
 
-  ///Fetches the number of merchants verified by the agent signed in [agent]
-  ///using google spanner APIs.
+  /// Fetches the number of merchants verified by the agent signed in [agent]
+  /// using google spanner APIs.
   int getMerchantVerified(){
```
Comment (krati-tiwari) [inline frontend/gpay_agent/lib/views/welcome_agent.dart]: Rename to getNumberOfMerchantsVerified

File: frontend/gpay_agent/lib/datamodels/agent.dart [hunk 2]
```diff
@@ -1,17 +1,27 @@
...
     this.AgentLastName,
     this.AgentPhone,
-    this.AgentDeviceID,
+    this.AgentLatitude,
+    this.AgentLongitude
   );
+
+  @override
+  String toString() {
+    return 'Agent{AgentEmail: $AgentEmail, AgentCreationDateTime: $AgentCreationDateTime, AgentFirstName: $AgentFirstName, AgentMidName: $AgentMidName, AgentLastName: $AgentLastName, AgentPhone: $AgentPhone, AgentLatitude: $AgentLatitude, AgentLongitude: $AgentLongitude}';
```
Comment (krati-tiwari) [inline frontend/gpay_agent/lib/datamodels/agent.dart]: split into multiple lines for better readability

File: frontend/gpay_agent/lib/main.dart [hunk 3]
```diff
@@ -16,7 +18,56 @@ class AgentApp extends StatelessWidget {
...
+  }
+
+  ///Initializes _goelocator and stream for listening changes in device location.
+  ///Position will be changed if distance changes by [distanceFilter]
+  @override
+  void initState() {
+    super.initState();
+
+    _geolocator = Geolocator();
+    LocationOptions locationOptions = LocationOptions(accuracy: LocationAccuracy.high, distanceFilter: 1);
```
Comment (krati-tiwari) [inline frontend/gpay_agent/lib/main.dart]: I would recommend storing this in a constant int member of class like UPDATE_DISTANCE

**Review by krati-tiwari — CHANGES_REQUESTED** [review]

File: frontend/gpay_agent/lib/datamodels/agent.dart [hunk 4]
```diff
@@ -1,17 +1,27 @@
 ///Model class for storing agent details.
 class Agent{
-  int AgentID;
+  String AgentEmail;
+  String AgentCreationDateTime;   //Look into the datatype
```
Comment (krati-tiwari) [inline frontend/gpay_agent/lib/datamodels/agent.dart]: Use Timestamp used by vahini?

Comment (Pragya1805) [reply frontend/gpay_agent/lib/datamodels/agent.dart]: Time stamp not being used in agent front end app, so leaving it as String

**Review by krati-tiwari — APPROVED** [review]

**Pull request merged**